NewHaven/src/TurnObserver.cpp

```diff
@@ -0,0 +1,150 @@
+    cout << msg << *GameController::current->game_settings->board->last_played->position <<  "!" << endl;
+    cout << "\n***UPDATED GAME BOARD CONTENT***" << endl;
+    GameController::current->game_settings->board->printBoard();
+    cout << "The resource marker has increased!" << endl;
+    GameController::current->game_settings->tracker->printScore();
+}
+
+
+void TurnObserver::printBV() {
+    string msg = "\n" + current->getID() + " is building their village!";
+    cout << msg << endl;
+}
+
+void TurnObserver::printSW() {
+    Player * sw_player = (*GameController::current->game_settings->players)[ GameController::current->getCurrentSharePlayer()];
+    string msg = "\n" + current->getID() + " is sharing their wealth with " + sw_player->getID();
+    cout << msg << endl;
+}
+void TurnObserver::printBuildingPlayed(bool flipped){
+    if(flipped)
+       cout << "\n" + current->getID() << " has played a building of type " <<
+                current->getVillage().last_played->building->getBuildingType() << " flipped at position "
+                << *current->getVillage().last_played->position
+                << "\n"  <<  *current->getVillage().last_played->vCost << " "
+                << current->getVillage().last_played->building->getBuildingType() << " has been consumed!"
+                << endl;
+    else
+        cout << "\n" + current->getID() << " has played a building of type " <<
+             current->getVillage().last_played->building->getBuildingType() << " at position "
+             << *current->getVillage().last_played->position
+             << "\n"  <<  *current->getVillage().last_played->vCost << " "
+             << current->getVillage().last_played->building->getBuildingType() << " has been consumed and removed from the resource tracker!"
+             << endl;
+
+}
+void TurnObserver::printBuildingPlayedSW(bool flipped){
+
+    Player * sw_player = (*GameController::current->game_settings->players)[ GameController::current->getCurrentSharePlayer()];
+    if(flipped)
+        cout << "\n" + sw_player->getID() << " has played a building of type " <<
+        sw_player->getVillage().last_played->building->getBuildingType() << " flipped at position "
+             << *sw_player->getVillage().last_played->position
+             << "\n"  <<  *sw_player->getVillage().last_played->vCost << " "
+             << sw_player->getVillage().last_played->building->getBuildingType() << " has been consumed!"
+             << endl;
+    else
+        cout << "\n" + sw_player->getID() << " has played a building of type " <<
+        sw_player->getVillage().last_played->building->getBuildingType() << " at position "
+             << *sw_player->getVillage().last_played->position
+             << "\n"  <<  *sw_player->getVillage().last_played->vCost << " "
+             << sw_player->getVillage().last_played->building->getBuildingType() << " has been consumed and removed from the resource tracker!"
+             << endl;
+
+}
```

Comment:
This can be very easily made into one function. It just need a way to distinguish whether it was called for share the wealth or for normal village building. A boolean parameter to the function could make that distinction. This isn't urgent though, don't waste time changing this.

Reply:
it does, but it wont be fixing it as it is just a niche thing for consistency